third_party/terraform/utils/provider_test.go.erb

```diff
@@ -196,6 +201,86 @@ func getTestAccProviders(testName string) map[string]terraform.ResourceProvider
 	}
 }
 
+// Wrapper for resource.Test to swap out providers for VCR providers and handle VCR specific things
+// Can be called when VCR is not enabled, and it will behave as normal
+func vcrTest(t *testing.T, c resource.TestCase, destroyFuncProducer func(provider *schema.Provider) func(s *terraform.State) error) {
+	providers := getTestAccProviders(t.Name())
+	c.Providers = providers
+	defer closeRecorder(t)
+	c.CheckDestroy = destroyFuncProducer(providers["google"].(*schema.Provider))
+	resource.Test(t, c)
+}
+
+// Produces a rand.Source for VCR testing based on the given mode.
+// In RECORDING mode, generates a new seed and saves it to a file, using the seed for the source
+// In REPLAYING mode, reads a seed from a file and creates a source from it
+func vcrSource(t *testing.T, path, mode string) (rand.Source, error) {
+	if s, ok := sources[t.Name()]; ok {
+		return s, nil
+	}
+	fileName := filepath.Join(path, fmt.Sprintf("%s.seed", t.Name()))
+	switch mode {
+	case "RECORDING":
+		seed := rand.Int63()
```

Comment:
nit - would like to see some comments/newlines or helper functions for some of the file stuff here, since personally I find it hard to tell what's supposed to be happening with IO calls in the mix